src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -315,12 +324,23 @@ private TexlBinding(IBinderGlue glue, IExternalRuleScopeResolver scopeResolver,
 
         // Binds a Texl parse tree.
         // * resolver provides the name context used to bind names to globals, resources, etc. This may be null.
-        public static TexlBinding Run(IBinderGlue glue, IExternalRuleScopeResolver scopeResolver, DataSourceToQueryOptionsMap queryOptionsMap, TexlNode node, INameResolver resolver, bool updateDisplayNames = false, DType ruleScope = null, bool forceUpdateDisplayNames = false, IExternalRule rule = null, bool useThisRecordForRuleScope = false)
+        public static TexlBinding Run(
+            IBinderGlue glue,
+            IExternalRuleScopeResolver scopeResolver,
+            DataSourceToQueryOptionsMap queryOptionsMap,
+            TexlNode node,
+            INameResolver resolver,
+            BindingConfig bindingConfig,
+            bool updateDisplayNames = false,
+            DType ruleScope = null,
+            bool forceUpdateDisplayNames = false,
+            IExternalRule rule = null,
+            bool useThisRecordForRuleScope = false)
```

Comment:
Can some of these other parameters also get moved to  BindingConfig? (maybe in a future PR).

Reply:
Yeah, that's the idea. This is an initial direction, but we should get most of these "How should this expression bind" switches moved over to the config object.

Reply:
2: Yes, that's the goal of this. 

1: Maybe? Still thinking about that. Would definitely make more sense. At least some of them can be grouped meaningfully into one set of flags.